srcjs/input_binding_fileinput.js

```diff
@@ -402,10 +397,13 @@ $.extend(fileInputBinding, {
       // (Chrome, Safari)
       $el.val("");
       el.files = e.originalEvent.dataTransfer.files;
+      // Recent versions of Firefox (57+, or "Quantum" and beyond) don't seem to
+      // automatically trigger a change event, so we trigger one manually here.
+      // On browsers that do trigger change, this operation appears to be
+      // idempotent, as el.files doesn't change between events.
+      $el.trigger("change");
```

Comment:
Can you confirm that this doesn't result in the file being uploaded twice?

Reply:
I made an app with this in my `server` function:
```R
observeEvent(input$file1, {
  cat("file uploaded\n")
})
```
I saw `file uploaded` in the R console only once per drop.